capabilities/writetarget/beholder/monitor/schema.go

```diff
@@ -0,0 +1,116 @@
+package monitor
+
+import (
+	"fmt"
+	"path"
+	"regexp"
+	"strings"
+
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/runtime/protoimpl"
+)
+
+const (
+	AttrKeyBeholderDataSchema = "beholder_data_schema"
+	AttrKeyBeholderEntity     = "beholder_entity"
+	AttrKeyBeholderDomain     = "beholder_domain"
+)
+
+// patternSnake is a regular expression to match CamelCase words
+// Notice: we use the Unicode property 'Lu' (uppercase letter) to match
+// the first letter of the word, and 'P{Lu}' (not uppercase letter) to match
+// the rest of the word.
+var patternSnake = regexp.MustCompile("(\\p{Lu}+\\P{Lu}*)")
+
+// toSnakeCase converts a CamelCase to snake_case (used for type -> file name mapping)
+func toSnakeCase(s string) string {
+	s = patternSnake.ReplaceAllString(s, "_${1}")
+	s, _ = strings.CutPrefix(strings.ToLower(s), "_")
+	return s
+}
+
+// toSchemaName returns a protobuf message name (short)
+func toSchemaName(m proto.Message) string {
+	return string(protoimpl.X.MessageTypeOf(m).Descriptor().Name())
+}
+
+// toSchemaName returns a protobuf message name (full)
+func toSchemaFullName(m proto.Message) string {
+	return string(protoimpl.X.MessageTypeOf(m).Descriptor().FullName())
+}
+
+// toSchemaPath maps a protobuf message to a Beholder schema path
+func toSchemaPath(m proto.Message, basePath string) string {
```

Comment:
This is something that is not directly tied to WT capability, and is a separate framework to emit Protobuf messages via Beholder. It should not be grouped under `/writetarget/...`

It potentially belongs in chainlink-common.

Reply:
Added TODO [here](https://github.com/smartcontractkit/chainlink-framework/blob/56c681ba2fc5fb260ef5ff3b2f1ac97b19ade00a/capabilities/writetarget/beholder/monitor/schema.go#L14) and compiled all TODOs into [this ticket ](https://smartcontract-it.atlassian.net/browse/BCFR-1222)